src/GrpAb/Map.jl

```diff
@@ -103,80 +103,106 @@ end
 #
 ################################################################################
 
-id_hom(G::FinGenAbGroup) = hom(G, G, identity_matrix(ZZ, ngens(G)), identity_matrix(ZZ, ngens(G)), check = false)
+@doc raw"""
+    identity_map(G::FinGenAbGroup) -> FinGenAbGroupHom
+
+Return the identity map of `G`.
+"""
+function identity_map(G::FinGenAbGroup)
+  I = identity_matrix(ZZ, ngens(G))
+  return FinGenAbGroupHom(G, G, I, I)::FinGenAbGroupHom
+end
 
 @doc raw"""
-    hom(A::Vector{FinGenAbGroupElem}, B::Vector{FinGenAbGroupElem}) -> Map
+    id_hom(G::FinGenAbGroup) -> FinGenAbGroupHom
 
-Creates the homomorphism $A[i] \mapsto B[i]$.
+Alias for [`identity_map`].
+"""
+id_hom(G::FinGenAbGroup) = identity_map(G)
+
+@doc raw"""
+    hom(
+      G::FinGenAbGroup,
+      H::FinGenAbGroup,
+      A::Vector{FinGenAbGroupElem},
+      B::Vector{FinGenAbGroupElem};
+      check::Bool=true,
+    ) -> FinGenAbGroupHom
+
+Return the group homomorphism $G\to H$ that sends `A[i]` to `B[i]` for all `i`.
+If `check` is set to `true`, the function checks whether the inputs indeed
+define a morphism of finitely generated abelian groups.
 """
-function hom(A::Vector{FinGenAbGroupElem}, B::Vector{FinGenAbGroupElem}; check::Bool = true)
-  GA = parent(A[1])
-  GB = parent(B[1])
-  @assert length(B) == length(A)
-  @assert length(A) > 0
-  #=
-  if (check)
-    m = vcat(ZZMatrix[x.coeff for x in A])
-    m = vcat(m, rels(parent(A[1])))
-    T = kernel(transpose(m), side = :right)
+function hom(
+    G::FinGenAbGroup,
+    H::FinGenAbGroup,
+    A::Vector{FinGenAbGroupElem},
+    B::Vector{FinGenAbGroupElem};
+    check::Bool=true
+  )
```

Comment:
Ideally, we want the types of `A` and `B` as stated. The method will complain if there is some type instable or sloppy code and we get a `Vector{Any}`. These errors can be confusing for beginners. 
Do we care? Maybe it would be better to just put `A::Vector` ? Any opinions?

Reply:
I agree. Just put `Vector`

Reply:
Sure